src/document/docUtil.ts

```diff
@@ -28,11 +28,12 @@ export function getPageNumber(element: Element | null): number | undefined {
 
 export function getRange(): Range | null {
     const selection = window.getSelection();
-    if (!selection || selection.type !== 'Range' || selection.isCollapsed) {
+    if (!selection || selection.type !== 'Range' || selection.isCollapsed || !selection.rangeCount) {
         return null;
     }
 
-    return selection.getRangeAt(0);
+    // return the last range
+    return selection.getRangeAt(selection.rangeCount - 1);
```

Comment:
How do multiple ranges get created?

Reply:
Use mouse to select texts while pressing "shift" (only possible on Firefox)

Reply:
Questions asked here will likely be asked by the next person looking at the code. It should be a cue to add a comment to the source.